Firewall Rule Based LED change

JonathanDLee24 opened the pull request:
This will light the led based on firewall rule number if a state is active set to what ever firewall ID you need. The example for me is 79



Ref:
https://forum.netgate.com/topic/182391/2100-led-question
https://forums.freebsd.org/threads/bin-sh-how-to-save-a-shell-command-output-into-a-string-variable.90088

JonathanDLee24 committed:
Create README_FIREWALLRULEBASEDLED

New code

JonathanDLee24 committed:
Create deviceonline.sh

deviceonline code bash

luckman212 commented:
This is a cool idea! I don't currently have a 2100 to test with anymore, but will probably have one in my posession in 2-3 weeks that I am getting back from a customer.

Either way, I'd like to see a little polishing up if we are going to merge it. It would be nice if the rule ID and color was configurable via script args or environment variables instead of the hardcoded `79`, for example.

JonathanDLee24 committed:
Update deviceonline.sh

Added three different versions.

JonathanDLee24 commented:
Thanks this was my first C type bash script. I agree it does need some cleaning up. I have it set to 79 as the rule ID but it could be anything even IP address based. The rules would have to be known by the user. pfctl -vvss would help users find the ID they want, and or pfctl -vvsr. 

For use with the scrip code:
pfctl -vvss| grep '192.168.1.11' would work great too as it would be IP address based not rule based, as well as..
pfctl -vvss| grep -e 192.168.1.18 -e 192.168.1.11 for multiple IP addresses

JonathanDLee24 committed:
Update deviceonline.sh

Added code to make a purple led,

also 4th version that uses the full output for use with multiple case commands to iterate over for multiple LED and device online use.

JonathanDLee24 commented:
Did you get to test this out? It glows red leds for game system use at night :) for me to catch my kid when he should be sleeping

JonathanDLee24 commented:
`#!/bin/sh
check_current_states=$( pfctl -vvss | grep -e ', rule 84' -e '192.168.1.11' -e '192.168.1.15' )
res=1
resb=1
resc=1
case "$check_current_states" in 
  *", rule 79"* ) res=0 ;;
esac
case "$check_current_states" in
  *192.168.1.11* ) resb=0 ;;
esac
case "$check_current_states" in
  *192.168.1.15* ) resc=0 ;;
esac
if [ $res = 0 ] && [ $resb = 0 ]; 
then
  sysctl -q dev.gpio.2.led.1.pwm=1
  gpioctl -f /dev/gpioc2 3 duty 50 >/dev/null
  sysctl -q dev.gpio.2.led.2.pwm=1
  gpioctl -f /dev/gpioc2 7 duty 0 >/dev/null
  gpioctl -f /dev/gpioc2 6 duty 50 >/dev/null
elif [ $res = 0 ];
then
  sysctl -q dev.gpio.2.led.1.pwm=1
  gpioctl -f /dev/gpioc2 3 duty 0 >/dev/null
  sysctl -q dev.gpio.2.led.2.pwm=1
  gpioctl -f /dev/gpioc2 7 duty 0 >/dev/null
  gpioctl -f /dev/gpioc2 6 duty 50 >/dev/null
elif [ $resb = 0 ];
then
  sysctl -q dev.gpio.2.led.2.pwm=1
  gpioctl -f /dev/gpioc2 7 duty 0 >/dev/null
  gpioctl -f /dev/gpioc2 6 duty 0 >/dev/null
  sysctl -q dev.gpio.2.led.1.pwm=1
  gpioctl -f /dev/gpioc2 3 duty 50 >/dev/null
else
  sysctl -q dev.gpio.2.led.1.pwm=1
  gpioctl -f /dev/gpioc2 3 duty 0 >/dev/null
  sysctl -q dev.gpio.2.led.2.pwm=1
  gpioctl -f /dev/gpioc2 6 duty 0 >/dev/null
  gpioctl -f /dev/gpioc2 7 duty 50 >/dev/null
  
fi
if [ $resc = 0 ];
then
  sysctl -q dev.gpio.2.led.0.pwm=1
  gpioctl -f /dev/gpioc2 2 duty 50 >/dev/null
else
  sysctl -q dev.gpio.2.led.0.pwm=1
  gpioctl -f /dev/gpioc2 2 duty 0 >/dev/null

fi`
